### PR TITLE
Added an in flight check to journal import requests

### PR DESCRIPTION
**Pull request by mjvkeulen**

### CODE REVIEW

File: src/django/VLE/models.py [hunk 1]
```diff
@@ -2196,6 +2196,14 @@ def get_update_response(self):
 
         return responses[self.state]
 
+    def target_url(self):
+        return '{}/Home/Course/{}/Assignment/{}/Journal/{}'.format(
+            settings.BASELINK, self.target.assignment.courses.first().pk, self.target.assignment.pk, self.target.pk)
+
+    def source_url(self):
+        return '{}/Home/Course/{}/Assignment/{}/Journal/{}'.format(
+            settings.BASELINK, self.source.assignment.courses.first().pk, self.source.assignment.pk, self.source.pk)
```
Comment (Larspolo) [inline src/django/VLE/models.py:2205]: the `gen_url` function is perfect for this use (should also be moved to the utils funciton, but that asside). `gen_url(journal=self.source)` will return this as well

Also, I cant seem to find where this is used, why did you add this?

Comment (mjvkeulen) [reply src/django/VLE/models.py:2205]: Use: just for debugging.

I couldn't remember the gen_url func, will just call that instead

**Review by Larspolo — APPROVED** [review]

File: src/vue/src/components/journal/JournalImportRequestApprovalModal.vue [hunk 2]
```diff
@@ -174,6 +176,8 @@ export default {
                     this.$root.$emit('bv::hide::modal', this.modalID)
                 }
                 this.$emit('jir-processed')
+            }).finally(() => {
+                this.jirPatchInFlight = false
```
Comment (Larspolo) [inline src/vue/src/components/journal/JournalImportRequestApprovalModal.vue:180]: Also for @engelhamer using this is way better than putting it both in the `catch` and `then`. Think we have a couple dozen places where this can also be improved, lets keep this trend going for new changes

**Review by Larspolo — APPROVED** [review]

**Review by engelhamer — APPROVED** [review]
